### ISSUE TITLE
Corrupted output image on Intel

### ISSUE THREAD

**Issue by werman** [DadSchoorse/vkBasalt#33]
I tested the latest master c6eb5f9b8706dfecd36a7d1e3f57631fd0043900 on UHD Graphics 630, 19.2.2:

![image](https://user-images.githubusercontent.com/3993671/68040712-8cafa400-fcd7-11e9-957b-d4987796de56.png)

<details>
 <summary>Validation layers complain on several things (Click me)</summary>

```

    #casSharpness specifies the amount of sharpning in the CAS shader.
    #0.0 less sharp, less artefacts, but not off
    #1.0 maximum sharp more artefacts
    #Everything in between in possible
    set option casSharpness equal to 0.4
    casSharpness=0.4

    #enableFxaa enables fxaa but currently turns off cas
    #0 fxxa off cas on
    #1 fxxa on  cas off
    set option enableFxaa equal to 0
    enableFxaa=0

    interrupted create instance
    set count 1
    before writing buffer descriptor Sets 
    0.4
    found graphic capable queue
    1
    queue 0x555772f4ef88
    queue 0x555772f4ef88
    queue 0x555772f4ef88
    format 50
    device 0x555772f4d1a0
    swapchain 0x16
    Interrupted create swapchain
    Interrupted get swapchain images 0
    Interrupted get swapchain images 3
    queue 0x555772f4ef88
    swapchain 0x16
    format 50
    device 0x555772f4d1a0
    fake image size: 8134676
    fake image alignment: 4096
    VUID-vkBindImageMemory-memoryOffset-01048(ERROR / SPEC): msgNum: 0 - vkBindImageMemory(): memoryOffset is 0x7c2014 but must be an integer multiple of the VkMemoryRequirements::alignment value 0x1000, returned from a call to vkGetImageMemoryRequirements with image. The Vulkan spec states: memoryOffset must be an integer multiple of the alignment member of the VkMemoryRequirements structure returned from a call to vkGetImageMemoryRequirements with image (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkBindImageMemory-memoryOffset-01048)
        Objects: 1
            [0] 0x18, type: 10, name: NULL
    VUID-vkBindImageMemory-memoryOffset-01048(ERROR / SPEC): msgNum: 0 - vkBindImageMemory(): memoryOffset is 0xf84028 but must be an integer multiple of the VkMemoryRequirements::alignment value 0x1000, returned from a call to vkGetImageMemoryRequirements with image. The Vulkan spec states: memoryOffset must be an integer multiple of the alignment member of the VkMemoryRequirements structure returned from a call to vkGetImageMemoryRequirements with image (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkBindImageMemory-memoryOffset-01048)
        Objects: 1
            [0] 0x19, type: 10, name: NULL
    after createFakeSwapchainImages 
    before creating swapchain image views 
    VUID-VkImageViewCreateInfo-usage-02275(ERROR / SPEC): msgNum: 0 - vkCreateImageView(): pCreateInfo->format VK_FORMAT_B8G8R8A8_SRGB with tiling VK_IMAGE_TILING_OPTIMAL does not support usage that includes VK_IMAGE_USAGE_STORAGE_BIT. The Vulkan spec states: If usage contains VK_IMAGE_USAGE_STORAGE_BIT, then the image view's format features must contain VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkImageViewCreateInfo-usage-02275)
        Objects: 1
            [0] 0x1b, type: 10, name: NULL
    VUID-VkImageViewCreateInfo-usage-02275(ERROR / SPEC): msgNum: 0 - vkCreateImageView(): pCreateInfo->format VK_FORMAT_B8G8R8A8_SRGB with tiling VK_IMAGE_TILING_OPTIMAL does not support usage that includes VK_IMAGE_USAGE_STORAGE_BIT. The Vulkan spec states: If usage contains VK_IMAGE_USAGE_STORAGE_BIT, then the image view's format features must contain VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkImageViewCreateInfo-usage-02275)
        Objects: 1
            [0] 0x1c, type: 10, name: NULL
    VUID-VkImageViewCreateInfo-usage-02275(ERROR / SPEC): msgNum: 0 - vkCreateImageView(): pCreateInfo->format VK_FORMAT_B8G8R8A8_SRGB with tiling VK_IMAGE_TILING_OPTIMAL does not support usage that includes VK_IMAGE_USAGE_STORAGE_BIT. The Vulkan spec states: If usage contains VK_IMAGE_USAGE_STORAGE_BIT, then the image view's format features must contain VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkImageViewCreateInfo-usage-02275)
        Objects: 1
            [0] 0x1d, type: 10, name: NULL
    before creating framebuffers 
    before creating fake swapchain image views 
    before creating descriptor Pool 
    set count 3
    after creating descriptor Pool 
    before allocating descriptor Sets 1
    before writing descriptor Sets 
    before writing descriptor Sets 
    before writing descriptor Sets 
    before creating descriptor Pool 
    0x7f76d7562b00
    0x7f76d7562b00
    0x555772ffc778
    0x555772e7f520
    0x555773005450
    0x55577300b1c0
    after allocateCommandBuffer 
    0x555772e7f520
    0x555773005450
    0x55577300b1c0
    VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184(ERROR / SPEC): msgNum: 0 - vkCmdPipelineBarrier(): pImageMemBarriers[0].srcAccessMask (0x20) is not supported by srcStageMask (0x400). The Vulkan spec states: Each element of pMemoryBarriers, pBufferMemoryBarriers and pImageMemoryBarriers must not have any access flag included in its srcAccessMask member if that bit is not supported by any of the pipeline stages in srcStageMask, as specified in the table of supported access types. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184)
        Objects: 1
            [0] 0x555772e7f520, type: 6, name: NULL
    VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184(ERROR / SPEC): msgNum: 0 - vkCmdPipelineBarrier(): pImageMemBarriers[0].srcAccessMask (0x20) is not supported by srcStageMask (0x400). The Vulkan spec states: Each element of pMemoryBarriers, pBufferMemoryBarriers and pImageMemoryBarriers must not have any access flag included in its srcAccessMask member if that bit is not supported by any of the pipeline stages in srcStageMask, as specified in the table of supported access types. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184)
        Objects: 1
            [0] 0x555773005450, type: 6, name: NULL
    VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184(ERROR / SPEC): msgNum: 0 - vkCmdPipelineBarrier(): pImageMemBarriers[0].srcAccessMask (0x20) is not supported by srcStageMask (0x400). The Vulkan spec states: Each element of pMemoryBarriers, pBufferMemoryBarriers and pImageMemoryBarriers must not have any access flag included in its srcAccessMask member if that bit is not supported by any of the pipeline stages in srcStageMask, as specified in the table of supported access types. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184)
        Objects: 1
            [0] 0x55577300b1c0, type: 6, name: NULL
    0writen commandbuffer0x555772e7f520
    1writen commandbuffer0x555773005450
    2writen commandbuffer0x55577300b1c0

```

</details>

**Comment by werman** [comment]
And debug mesa just crashes on `CreateImageView`  after layers complain about 
```vkCreateImageView(): pCreateInfo->format VK_FORMAT_B8G8R8A8_SRGB with tiling VK_IMAGE_TILING_OPTIMAL does not support usage that includes VK_IMAGE_USAGE_STORAGE_BIT```

**Comment by DadSchoorse** [comment]
Oh I missed that on the rewrite... We do not need `VK_IMAGE_USAGE_STORAGE_BIT` anymore.
I have fixed the validation layer messages now.

**Comment by DadSchoorse** [comment]
But I have not fix the first error... That one is annoying, because I assumed that the size is aligned, which seems to be the case on NVIDIA and AMD.

**Comment by DadSchoorse** [comment]
The memory error should be fixed by 4c4020479d47809c2ba225840933323a00e3d1c8

**Comment by werman** [comment]
This fixes the issue. vkcube and VkQuake work correctly.
